helperFunctions/SalesforceAPIHelpers.ts

```diff
@@ -127,6 +97,40 @@ export async function sendChatRequest(
 		sendChatRequestHttpRequest.data.prechatDetails.push(customDetailJSON);
 	}
 
+	if (prechatDetails) {
+		sendChatRequestHttpRequest.data.prechatDetails.push(JSON.parse(prechatDetails));
+	} else {
+		sendChatRequestHttpRequest.data.prechatDetails.push({
```

Comment:
Is this to support both `customDetail` and `prechatDetails`. I think we only need one?

In fact, I think if DF returns all the fields in one(transcript, email, salesforce case id, appVersion) as an array it should work.

So after we are done with changes from the DF side we should remove `customDetail` and only maintain`prechatDetails`

Reply:
No actually we need both to have backward compatibility. Not every time will we get the pre chat details for CX agent